tests/service/messaging/mqtt/test_send_direct_message_service.py

```diff
@@ -1,51 +1,128 @@
-    sender = OnboardResponseIntegrationService.read(SENDER)
+    """
+    Test to send the message to a recipient
+    The existing sender and recipient PEM onboard responses are read using OnboardIntegrationService
+    """
+    _recipient_onboard_response = OnboardResponseIntegrationService.read(Identifier.MQTT_RECIPIENT_PEM[Identifier.PATH])
+    _sender_onboard_response = OnboardResponseIntegrationService.read(Identifier.MQTT_SENDER_PEM[Identifier.PATH])
+
+    @staticmethod
+    def test_given_valid_message_content_when_sending_message_to_single_recipient_then_the_message_should_be_delivered():
+        """
+        Test for sending the valid message content to a single recipient after enabling IMG_PNG capability with sender and recipient
+        Open Connection between Recipient and Agrirouter is required. The setup between the sender and the recipient is done before
+        running the test. If
+        """
+        TestSendDirectMessageService._enable_recipient_capabilities_via_mqtt()
+
+        current_sequence_number = SequenceNumberService.generate_sequence_number_for_endpoint(TestSendDirectMessageService._recipient_onboard_response.get_sensor_alternate_id())
+
+        send_message_parameters = SendMessageParameters(
+            onboarding_response=TestSendDirectMessageService._sender_onboard_response,
+            technical_message_type=CapabilityType.IMG_PNG.value,
+            application_message_id=new_uuid(),
+            application_message_seq_no=current_sequence_number,
+            recipients=[TestSendDirectMessageService._recipient_onboard_response.get_sensor_alternate_id()],
+            base64_message_content=DataProvider.read_base64_encoded_image(),
+            mode=RequestEnvelope.Mode.Value("DIRECT"))
+
+        messaging_service = MqttMessagingService(
+            onboarding_response=TestSendDirectMessageService._sender_onboard_response,
+            on_message_callback=TestSendDirectMessageService._on_message_callback)
+
+        send_message_service = SendMessageService(messaging_service=messaging_service)
+        send_message_service.send(send_message_parameters)
+        Sleeper.let_agrirouter_process_the_message(seconds=5)
+
 
-    def set_capabilities_for_sender(self):
-        capabilities_service = CapabilitiesService(messaging_service=HttpMessagingService())
+    @staticmethod
+    def _on_message_callback(client, userdata, msg):
+        """
+        Callback to handle the incoming messages from the MQTT broker
+        """
+        outbox_message = OutboxMessage()
+        outbox_message.json_deserialize(msg.payload.decode().replace("'", '"'))
+        decoded_message = decode_response(outbox_message.command.message.encode())
+        while not decoded_message:
+            Sleeper.let_agrirouter_process_the_message(seconds=5)
+
+        assert type(decoded_message.response_payload.details.value) == bytes
+        assert decoded_message.response_payload.details.value is not None
```

Comment:
Compare with a dedicated value, i.e. something like a hash to ensure that this is the same value.

Reply:
The `value.details` contains the content of the message, means the image or the ZIP.

https://github.com/DKE-Data/agrirouter-api-protobuf-definitions/blob/main/java/src/main/resources/com/dke/data/agrirouter/proto/messaging/response/response.proto

Reply:
Yes, it contains the content but also the header. If I try to compare the hashed value of the input data image, and the received content, it won't be the same even though the received content contains the same image. This is because the byte string received is different to the one sent.